### ISSUE TITLE
client object uri method unnecessarily expensive

### ISSUE THREAD

**Issue by ysth** [rustyconover/net-amazon-s3#60]
In 0.85, the following ends up making an http request in Net::Amazon::S3::Bucket::_head_region, when as far as I can tell there is absolutely no reason to, and the uri should be able to be formed without such a request:

```
my $s3 = Net::Amazon::S3->new(aws_access_key_id=>$key,aws_secret_access_key=>$secret,retry=>1,use_virtual_host=>1,authorization_method=>"Net::Amazon::S3::Signature::V4");
my $client = Net::Amazon::S3::Client->new(s3=>$s3);
my $bucket = $client->bucket(name=>$bucket,region=>$region);
my $uri = $bucket->object(key=>$key)->uri;
```

**Comment by happy-barney** [comment]
True, that HEAD request is not necessary once you construct bucket with region.

**Comment by happy-barney** [comment]
Fixed in #63
